src/vos/tests/vos_size_dfs_sample.c

```diff
@@ -0,0 +1,292 @@
+			fprintf(fp, "  value_type: single_value\n");
+		}
+		if (iods[i].iod_type == DAOS_IOD_ARRAY) {
+			fprintf(fp, "  value_type: array\n");
+		}
+
+		fprintf(fp, "  values: [{\"count\": %u, \"size\": %u}]\n",
+			iods[i].iod_nr, (unsigned int)iods[i].iod_size);
+		fprintf(fp, "\n");
+	}
+
+	/* print the D-Key value */
+	to_lower(buf, dkey->iov_buf, dkey->iov_len);
+
+	fprintf(fp, "%s: &%s\n", buf, buf);
+	fprintf(fp, "  size: %zu\n", dkey->iov_buf_len);
+	fprintf(fp, "  overhead: meta\n");
+	print_list(fp, "akeys", values, akey_count);
+	fprintf(fp, "\n");
+}
+
+/**
+* TODO: add option to specify the number of files and its size
+*/
+void
+print_dfs_example_remainder(FILE *fp, int dfs_inode_size)
+{
+	fprintf(fp,
+	"dfs_inode: &dfs_inode\n"
+	"  type: integer\n"
+	"  overhead: meta\n"
+	"  value_type: array\n"
+	"  values: [{\"count\": 1, \"size\": %d}]\n"
+	"\n"
+	"# Assumes 16 bytes for file name\n"
+	"dirent_key: &dirent\n"
+	"  count: 1000000\n"
+	"  size: 16\n"
+	"  akeys: [*dfs_inode]\n"
+	"\n"
+	"dir_obj: &dir\n"
+	"  dkeys: [*dirent]\n"
+	"\n"
+	"superblock: &sb\n"
+	"  dkeys: [*dfs_sb_metadata]\n"
+	"\n"
+	"array_akey: &file_data\n"
+	"  size: 1\n"
+	"  overhead: meta\n"
+	"  value_type: array\n"
+	"  values: [{\"count\": 1, \"size\": 4096}]\n"
+	"\n"
+	"array_meta: &file_meta\n"
+	"  size: 19\n"
+	"  overhead: meta\n"
+	"  value_type: single_value\n"
+	"  values: [{\"size\": 24}]\n"
+	"\n"
+	"file_dkey_key0: &file_dkey0\n"
+	"  count: 1\n"
+	"  type: integer\n"
+	"  akeys: [*file_data, *file_meta]\n"
+	"\n"
+	"file_dkey_key: &file_dkey\n"
+	"  count: 1\n"
+	"  type: integer\n"
+	"  akeys: [*file_data]\n"
+	"\n"
+	"file_key: &file\n"
+	"  count: 1000000\n"
+	"  dkeys: [*file_dkey0, *file_dkey]\n"
+	"\n"
+	"posix_key: &posix\n"
+	"  objects: [*sb, *file, *dir]\n"
+	"\n"
+	"containers: [*posix]\n", dfs_inode_size);
+}
+
+int
+main(int argc, char **argv)
+{
+	FILE				*fp;
+	char				*fname = NULL;
+	daos_iod_t			*akey_sb;
+	daos_key_t			dkey_sb;
+	static struct option		long_options[] = {
+		{"fname",		required_argument, 0, 'f'},
+		{"help",		no_argument, 0, 'h'},
+	};
+
+	int				 rc;
+	int				 index = 0;
+	int				 opt = 0;
+	int				 akey_count;
+	int				 dfs_inode_size;
+
+	rc = daos_debug_init(NULL);
+	if (rc) {
+		printf("Error initializing debug system\n");
+		return rc;
+	}
+
+	while ((opt = getopt_long(argc, argv, "a:f:h",
+				  long_options, &index)) != -1) {
+		switch (opt) {
+		case 'f':
+			fname = alloc_fname(optarg);
+			if (fname == NULL)
+				goto exit_1;
+			break;
+		case 'h':
+			print_usage(argv[0]);
+			goto exit_1;
+		default:
+			printf("Unknown option\n");
+			print_usage(argv[0]);
+			goto exit_1;
+		}
+	}
+
+	fp = open_file(fname);
+	free_fname(fname);
+
+	if (fp == NULL)
+		goto exit_1;
+
+	rc = get_sb_layout(&dkey_sb, &akey_sb, &akey_count, &dfs_inode_size);
+	if (rc) {
+		goto exit_1;
+	}
+
+	fprintf(fp, "---\n"
+		"# Sample conflig file DFS files and directories\n"
+		"num_pools: 1000\n\n");
```

Comment:
what is num_pools: 1000?

Reply:
I copy-pasted that value from our current implementation of [vos_dfs_sample.yaml](https://github.com/daos-stack/daos/blob/master/src/vos/tests/vos_dfs_sample.yaml) 
I am working on a tool (DAOS-4605) to create this kind of yaml files from user input data (like CSV files). I think that I can add a parameter to define the number of pool on that PR.

Reply:
num_pools should probably be named num_shards.   It's the count of vos pools

Reply:
@jolivier23 Should I re-name ( s/num_pools/num_shards/ ) on [vos_size_input.yaml](https://github.com/daos-stack/daos/blob/master/src/vos/tests/vos_size_input.yaml) and[ vos_size.py](https://github.com/daos-stack/daos/blob/master/src/vos/tests/vos_size.py) as well?